packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -31,23 +31,40 @@ interface SelectPanelMultiSelection {
   onSelectedChange: (selected: ItemInput[]) => void
 }
 
+type OpenGestures = 'anchor-click' | 'anchor-key-press'
+type CloseGestures = 'click-outside' | 'escape' | 'selection' | 'save-click' | 'cancel-click'
+
 interface SelectPanelBaseProps {
   // TODO: Make `title` required in the next major version
   title?: string | React.ReactElement
   subtitle?: string | React.ReactElement
-  onOpenChange: (
-    open: boolean,
-    gesture: 'anchor-click' | 'anchor-key-press' | 'click-outside' | 'escape' | 'selection',
-  ) => void
+  onOpenChange: (open: boolean, gesture: OpenGestures | CloseGestures) => void
   placeholder?: string
   // TODO: Make `inputLabel` required in next major version
   inputLabel?: string
   overlayProps?: Partial<OverlayProps>
   footer?: string | React.ReactElement
 }
 
+type SelectPanelVariantProps =
+  | {
+      // do we need keep the old variants for backward-compat?
+      // there isn't any usage in dotcom
+      variant: 'anchored' | FilteredActionListProps['variant']
```

Comment:
>do we need keep the old variants for backward-compat? there isn't any usage in dotcom

I think we should. Although I am not sure if should merge the old variants and the new ones since they seem different variants to me. The new one is more of a type (modal, anchored) and the other (full and inset ) feels related to the stylings of the panel. 

One idea might be (For the sake of keeping things uncomplicated) we can merge them and immediately deprecated the old variants?

Curious to hear your thoughts

Reply:
> Although I am not sure if should merge the old variants and the new ones since they seem different variants to me

That's very true.

I personally would like to get rid of the old ones because it's a design changing option that has _leaked out_ of deprecated ActionList all the way to SelectPanel. I don't think we want folks to use the full width with SelectPanel.

> One idea might be (For the sake of keeping things uncomplicated) we can merge them and immediately deprecated the old variants?

I like that

Luckily, there is no usage in dotcom and only [5 instances in internal status page](https://primer-query.githubapp.com/?name=selectpanel&attribute=variant&entrypoint=%22%40primer%2Freact%22), so it feels like we can remove them eventually.